### ISSUE TITLE
Shopfront search box returns all records when none is expected

### ISSUE THREAD

**Issue by filipefurtad0** [openfoodfoundation/openfoodnetwork#5467]
## Description

The search box in the shopfront (/shops) is showing all records when it should return zero results. This appears to be introduced by the latest release 2.9.9 #5433 .

## Expected Behavior

No results should be returned if there are no shops matching the text inserted in the search box.

## Actual Behaviour

All the shops are being returned.

## Steps to Reproduce

1. Go to /shops
2. Insert a string in the search box which does not match any criteria in the shop list
3. Observe all results being returned

## Animated Gif/Screenshot

The video from @luisramos0 shows the behaviour of the search box on a staging server with release 2.9.9 deployed, and compares it with a production server:

![Peek 2020-05-19 11-27](https://user-images.githubusercontent.com/49817236/82328167-6a614780-99d7-11ea-9ec9-b66a0e5cbae6.gif)

## Workaround

The search box can still be used, but only when the input partially matches the existing shops.

## Severity

bug-s4: it's annoying, but you can use it

## Your Environment

* Version used: 2.9.9.
* Browser name and version: Firefox 75
* Operating System and version (desktop or mobile): Desktop

**Comment by luisramos0** [comment]
Hmm.. I cant see what's going on here so I tried findind out what PR introduced this, no luck yet.
I can see that both:
https://github.com/openfoodfoundation/openfoodnetwork/pull/5156
and
https://github.com/openfoodfoundation/openfoodnetwork/pull/5356
did not cause this bug...
@Matt-Yorkley any idea?

**Comment by kirstenalarsen** [comment]
as an s3 why is this in dev-ready when no instance has nominated it? Not sure if I am missing something but moving it back to all the things - happy to be overruled @filipefurtad0 @luisramos0 @RachL @lin-d-hop

**Comment by kirstenalarsen** [comment]
I agree it is horrible - but to use the process we need an instance to select it as it's most horrible before it can cut in front of all the others

**Comment by RachL** [comment]
@kirstenalarsen you validated yourself the position a few days ago: https://openfoodnetwork.slack.com/archives/CAVTM01QB/p1590060958274400 :-) so you have change your mind about it?

**Comment by luisramos0** [comment]
yes, we agreed we would fix these regressions.
I am ok if you want to change that decision :+1:

**Comment by luisramos0** [comment]
and I am sorry for not mentioning the slack conversation behind this in the first place:
![image](https://user-images.githubusercontent.com/1640378/83510144-6dbefd80-a4c4-11ea-8da0-ef6060c22eb8.png)

**Comment by jeduardo824** [comment]
Hey guys, I opened a PR with a suggested fix. The showed shops above are the "distance matches" shops. We can erase them too, but since I don't know the expected behavior from the product view, I only fixed to show the message "No results found."

**Comment by Matt-Yorkley** [comment]
There's various cases where the search can return results if the term doesn't match and enterprise name. Like entering a location gives results based on distance from that location. I think with partial matches it will return a set of results that look similar....

I've just had a play around with this in production though, and it definitely looks a bit weird.
